app/javascript/src/component_expander.js

```diff
@@ -47,80 +64,101 @@ const {
 **/
 class Expander {
   #config;
+  #state;
 
   constructor($node, config) {
     const conf = mergeObjects({
-      title_tag: 'h2',
-      wrap_content: false,
+      title: null,
```

Comment:
Yeah, I wasn't sold on the name title, the problem with activator is that this results in the use case that we are actually using being really confusing to read.  

We would take in `conf.activator`, create a new actual `$activator`, wrap the inner contents of `conf.activator` with `$activator` and then set `this.$activator` to `$activator` 😱 

tbh this was sort of why I did this whole refactor in the first place  - this whole having an 'activator' passed in that isn't really the activator at all was what I was attempting to avoid!

Reply:
Well you've probably just highlighted why having the choice of two was probably easier to understand. 
`config.activator_text`
or
`config.activator`

Previously we also had the simplicity that if you passed an element we just enhanced that element and left in place or, if one was to be created (from activator_text) then it would be injected into DOM as previous sibling to expanding/collapsing content (_think_ but not checked actual code). The end result could be better anticipated with little knowledge/thinking (that helps my brain :-) )

If we're going for a single config option, I'd still prefer something that at least refers to or indicates some relationship with the resulting activator. Title immediately indicates something entirely different and unrelated (due to common usage for many other things). 

Perhaps a rename from conf.activator to something like conf.activator_source or conf.activator_target ??
Think I'd prefer 'source' there as that's closest to how it is being used, and maybe not something as verbose config.activator_text_or_element_to_provide_text_or_a_button :-) 

As for handling the new preferred structure of `<button>` inside `<h2>`, I think it needn’t get too complicated or even possibly need to expect anything specific .  As I see it, with the new changes proposed, there should now be three rules.

If `conf.activator_source` is:
1. a `String` then create a button and place as previous sibling to $node container
2. an `Element` (container tag only) create a button from container text and place button in element
3. a `Button` element or `Empty element` then just enhance it with required functionality

 I'd probably ignore a couple of potential issues there 
e.g.
- If the container tag has other elements and/or more text than wanted
- When it's an Empty element, since it kind of defeats the whole point of change, maybe code just includes a dev comment about skipped support, leaving rule 3 as buttons only. 

In other words, those potential issues could probably be mentally filed under "future developer issue and/or maybe they should rethink how to apply their activator".


So, from a code perspective, we now might have something like: 
```
// in constructor...
var $button = createActivator(conf.activator); // figures out if not button needs to create and how, etc.
this.$activator = enhanceActivator($button); // adds functionality and links up with the Expander content, etc. 

```
With…
  ``` 
  // Takes whatever config value, returns a button
createActivator(thing) {
  var $activator = thing; // in case already a button
  IF thing is String
      $activator = create button with thing as text
      place button before content container
  ELSE
      IF thing has text and is not a button 
          $activator = create button from thing text 
          place button inside thing  
          
   return $activator
}

// Takes a button (or empty, if supporting) element, enhances with events + Aria, returns enhanced element 
enhanceActivator($activator) {
    $activator.attr(“Aria”, blah); 
    $activator.onclick… 
    return $activator
 }
```

Reply:
So, I only spotted this comment after applying my own solution.  D'oh.  

I think I've mostly done things as above.  the main difference is that I changed the config property from `title` to `activator` (as suggested).  Then to avoid naming confusion internally, I renamed `$activator` in the class to be `$button` which is pretty specific as that is what this whole change mostly does - ensure that the activator is always a `<button>` element.

I've also written tests too. Hopefully one less thing on your plate - or if you feel like there are improvements/tweaks you could always address them when you get to this one.